src/NHibernate/Logging.cs

```diff
@@ -85,16 +159,16 @@ private static INHibernateLoggerFactory GetLoggerFactory(string nhibernateLogger
 			var loggerFactoryType = System.Type.GetType(nhibernateLoggerClass);
 			try
 			{
-				object loadedLoggerFactory = Activator.CreateInstance(loggerFactoryType);
-#pragma warning disable 618
-				if (loadedLoggerFactory is ILoggerFactory oldStyleFactory)
+				var loadedLoggerFactory = Activator.CreateInstance(loggerFactoryType);
+				if (loadedLoggerFactory is INHibernateLoggerFactory newStyleFactory)
```

Comment:
Test inverted because log4net logger factory does now implement both interfaces (old and new).